question28/js/login.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+import { validationInfo } from "./validation";
+import { Chance } from "chance";
+const chance = new Chance();
+
+const submitButton = document.getElementById('js-submit-button');
+const userInfoInputArea = document.getElementById('name_mail');
+const passwordInputArea = document.getElementById('password');
+const form = document.getElementById('form');
+
+if(localStorage.getItem('token')) {
+  window.location.href = "./index.html";
+}
+
+const isValidStatus = {
+  name_mail: false,
+  password: false,
+}
+
+const setValidationEvents = (isValid, targetForm) =>
+  isValid ? validEvent(targetForm) : invalidEvent(targetForm);
+
+const validEvent = (targetForm) => {
+  targetForm.classList.add("valid");
+  targetForm.classList.remove("invalid");
+  targetForm.nextElementSibling.textContent = "";
+  isValidStatus[targetForm.id] = true;
+};
+
+const invalidEvent = (targetForm) => {
+  targetForm.classList.add("invalid");
+  targetForm.classList.remove("valid");
+  targetForm.nextElementSibling.textContent =
+    validationInfo[targetForm.id].errorMessage;
+  isValidStatus[targetForm.id] = false;
+};
+
+
+const validateInputValue = (e) => {
+  const targetForm = e.target;
+  const value = targetForm.value.trim();
+  const result = validationInfo[targetForm.id].validation(value);
+  setValidationEvents(result,targetForm);
+  renderRequiredFieldMessages(targetForm);  
+  toggleDisabledOfSubmitButton(checkAllValidity());
+} 
+
+const checkAllValidity = () => {
+  return Object.values(isValidStatus).every((result) => result );
+}
+
+const toggleDisabledOfSubmitButton = (isValid) => {
+  submitButton.disabled = isValid ? false : true;
+} 
+
+const renderRequiredFieldMessages = (targetForm) => {
+  if (targetForm.value.trim() === "") {
+    targetForm.nextElementSibling.textContent = "※入力必須項目です";
+  }
+};
+
+userInfoInputArea.addEventListener('blur', validateInputValue);
+passwordInputArea.addEventListener('blur', validateInputValue);
+
+
+const checkData = ({ name_mail, user_password }) => {
+  const userJasonDataInServer = localStorage.getItem('userData');
+  const registeredUserData = JSON.parse(userJasonDataInServer);
```

Comment:
IMO
```suggestion
  const userJsonDataInServer = localStorage.getItem('userData');
  const registeredUserData = JSON.parse(userJasonDataInServer);
```

or

I think we can combine them into one.
``` js
 const registeredUserData = JSON.parse(localStorage.getItem('userData'));
```

Reply:
Thank you for your review! You are right! Fixed so please confirm. (There were two of these and both have been corrected.)

fixed commit: 
https://github.com/csakyo/homework/pull/42/commits/ea2c1a44399516747a7d6f778db14f03f10680d7
https://github.com/csakyo/homework/pull/42/commits/4872a6b05548fbbf9832b263062aa99b5029029d